File: backend/staff-service/src/main/java/com/tnl/staffservice/entity/Staff.java
package com.tnl.staffservice.entity;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.GenericGenerator;

import javax.persistence.*;
import java.time.LocalDate;
import java.util.Date;


@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "staff")
public class Staff {
    @Id
    @GeneratedValue(generator = "uuid")
    @GenericGenerator(name = "uuid", strategy = "uuid2")
    private String id;
    private String code;
    private String firstName;
    private String lastName;
    private String fullName;

    @Temporal(TemporalType.DATE)
    @JsonFormat(pattern="yyyy-MM-dd")
    private Date dateOfBirth;
    private Long branchId;
    private Long departmentId;
    private Long positionId;
    @Temporal(TemporalType.DATE)
    @JsonFormat(pattern="yyyy-MM-dd")
    private Date joinDate;
    private Integer status;

    public Staff(String code, String firstName, String lastName, Date dateOfBirth, Long branchId, Long departmentId, Long positionId, Date joinDate, Integer status) {
        this.code = code.trim();
        this.firstName = firstName;
        this.lastName = lastName;
        this.fullName = firstName + " " + lastName;
        this.dateOfBirth = dateOfBirth;
        this.branchId = branchId;
        this.departmentId = departmentId;
        this.positionId = positionId;
        this.joinDate = joinDate;
        this.status = status;
    }
}

File: frontend/bidcdashboard/lib/api/model/staff.dart
class Staff {
  final String id;
  final String code;
  final String firstName;
  final String lastName;
  // final String fullName;
  final dynamic dateOfBirth;
  final int branchId;
  final int departmentId;
  final int positionId;
  final dynamic joinDate;
  final int status;

  Staff(
      {required this.id,
      required this.code,
      required this.firstName,
      required this.lastName,
      // required  this.fullName,
      required this.dateOfBirth,
      required this.branchId,
      required this.departmentId,
      required this.positionId,
      required this.joinDate,
      required this.status});

  factory Staff.fromJson(Map<String, dynamic> json) {
    return Staff(
      id: json['id'],
      code: json['code'],
      firstName: json['firstName'],
      lastName: json['lastName'],
      dateOfBirth: json['dateOfBirth'],
      branchId: json['branchId'],
      departmentId: json['departmentId'],
      positionId: json['positionId'],
      joinDate: json['joinDate'],
      status: json['status'],
    );
  }

  Map<String, dynamic> toJson() => {
        'id': id,
        'code': code,
        'firstName': firstName,
        'lastName': lastName,
        'dateOfBirth': dateOfBirth,
        'branchId': branchId,
        'departmentId': departmentId,
        'positionId': positionId,
        'joinDate': joinDate,
        'status': status,
      };
}
